### PR TITLE
Revocation transactions' signatures exchange

### PR DESCRIPTION
**Pull request by darosior**
This builds on top of the recent Coordinator integration to implement the exchange of revocation signatures through it.

### Content

- This completely refactors the configuration. We'll soon have more keys and the former layout was not optimal. This also cleans up quite a lot src/common/config.rs. Then we integrate the Coordinator's information (host, Noise static public key)
- This improves (and refactor, again) the functional test framework to
  1. integrate and manage the Coordinator daemon
  2. Abstract out the stakeholders/managers, their wallets, and their keys
  3. Add a boutique signer module to the test suite (PSBT serialization, sighash computation, secp256k1 sigs via `coincurve`)
  Tried to add way more documentation about how to run the tests, especially with the new Postgre integration
- We also add a small handy tool in `contrib/` to hand-sign PSBTs with an xpriv
- We then implement `revocationtxs` for the stakeholders to push their signatures to the Coordinator.
  The signature fetching is done as a new background thread, for any profile: we now store unsigned PSBTs in the db at the deposit confirmation time, then update them with signatures as we go either in the background thread or via the `revocationtxs` RPC command.

Refer to the commit messages for specifics.

### A few pitfalls

- The signature encryption is on hold for now because of https://github.com/re-vault/practical-revault/issues/72 . So right now the managers can poll all the revocation transactions. Eventually, we should restrict them from storing the Emergency transactions and this would implicitly make the signature fetcher upgrade a vault to `Secured`  after receiving only the Cancel transaction sig for them.
- A few interfaces are mixed between "pre-signed transactions" (Unvault included) and "revocation transactions" (Unvault excluded). The Unvault signatures and vault upgrade to `Active` will however be able to use the same path.

### Notes

This builds on https://github.com/re-vault/revault_tx/pull/51 so please review that first

### CODE REVIEW

**Comment by darosior** [comment]
Ok the CI failure is that i didn't init the Coordinator submodule in the bash script. Should be fixed now.

**Comment by darosior** [comment]
Amended now that https://github.com/re-vault/revault_tx/pull/51 is merged

File: src/daemon/sigfetcher.rs [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            }
+            RevaultTx::Cancel(cancel_tx) => {
+                get_sigs(revaultd, tx.id, tx.vault_id, cancel_tx)?;
+            }
+            RevaultTx::Emergency(emer_tx) => {
+                get_sigs(revaultd, tx.id, tx.vault_id, emer_tx)?;
+            }
+            RevaultTx::UnvaultEmergency(unemer_tx) => {
+                get_sigs(revaultd, tx.id, tx.vault_id, unemer_tx)?;
+            }
```
[the remark this answers is not in the document]
Comment (darosior) [reply src/daemon/sigfetcher.rs:202]: Tried but can't find a neat way (`match` wants the same type on each arm!). Do you have an idea ? It's even crappier later in the db function

[the remark this answers is not in the document]
Comment (edouardparis) [reply src/daemon/sigfetcher.rs:202]: If we change a little bit revault_tx::RevaultTransaction, we can add to RevaultTx to implement it:

```rust
impl RevaultTransaction for RevaultTx {
    fn inner_tx(&self) -> &Psbt {
        match self {
            RevaultTx::Unvault(tx) => tx.inner_tx(),
            RevaultTx::Cancel(tx) => tx.inner_tx(),
            RevaultTx::Emergency(tx) => tx.inner_tx(),
            RevaultTx::UnvaultEmergency(tx) => tx.inner_tx(),
        }
    }

    fn inner_tx_mut(&mut self) -> &mut Psbt {
        match self {
            RevaultTx::Unvault(tx) => tx.inner_tx_mut(),
            RevaultTx::Cancel(tx) => tx.inner_tx_mut(),
            RevaultTx::Emergency(tx) => tx.inner_tx_mut(),
            RevaultTx::UnvaultEmergency(tx) => tx.inner_tx_mut(),
        }
    }
}
```
and pass RevaultTx to get_sigs
(but it is not really useful to do it in this PR)

[the remark this answers is not in the document]
Comment (darosior) [reply src/daemon/sigfetcher.rs:202]: What would we need to modify in `revault_tx` ?

[the remark this answers is not in the document]
Comment (edouardparis) [reply src/daemon/sigfetcher.rs:202]: remove `RevaultTransaction::from_psbt_serialized`

**Review by edouardparis — APPROVED** [review]
ACK 75a5e53a7deb1d02a405d8b02db8947dbc88b181

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,6 @@
+[submodule "tests/coordinatord"]
+	path = tests/coordinatord
+	url = https://github.com/re-vault/coordinatord
+[submodule "tests/servers/coordinatord"]
+	path = tests/servers/coordinatord
+	url = https://github.com/re-vault/coordinatord
```
Comment (darosior) [inline .gitmodules:6]: Argh i commited a garbage submodule, will remove..

**Comment by darosior** [comment]
ACK ab4e92e647178ade213f583828b2977ada95385a -- re-applying Edouard Paris' ACK

Only change with last force-push is the removal of a submodule i added by mistake. See the range diff:
```diff
$ git range-diff master origin/sig_sharing sig_sharing
 1:  a033367 !  1:  1b408cc qa: network support for functional tests framework
    @@ Commit message
     
      ## .gitmodules (new) ##
     @@
    -+[submodule "tests/coordinatord"]
    -+  path = tests/coordinatord
    -+  url = https://github.com/re-vault/coordinatord
     +[submodule "tests/servers/coordinatord"]
     +  path = tests/servers/coordinatord
     +  url = https://github.com/re-vault/coordinatord
    @@ tests/README.md
     +POSTGRES_USER="test" POSTGRES_PASS="test" TEST_DEBUG=1 pytest -vvv -n8
      ```
     
    - ## tests/coordinatord (new) ##
    -@@
    -+Subproject commit 48b67f72b948e680c78276d73b2e50dffbdddb7f
    -
      ## tests/fixtures.py ##
     @@ tests/fixtures.py: Most of the code here is stolen from C-lightning's test suite. This is surely
      Rusty Russell or Christian Decker who wrote most of this (I'd put some sats on
 2:  261db70 =  2:  266a3c4 ci: setup postgres instance for functional tests
 3:  5cab1f1 =  3:  28715e7 revaultd: read Noise static private key on startup.
 4:  2c07715 =  4:  90adf9d contrib: add a "PSBT signer" tool
 5:  886024a =  5:  ef0f7b2 revaultd: fix deposit descriptor pubkeys
 6:  b1dbf7a =  6:  fb2dbf5 tests/servers: update coordinatord
 7:  3e037c3 =  7:  35de0ad tests: add Andrew Chow's PSBT serialization in Python
 8:  a0f86cd =  8:  d75715e tests: add ANYONECANPAY sighash serialization to achow's psbt sers
 9:  cc6f220 =  9:  7e59d2b jsonrpc: restrict 'getrevocationtxs' and 'revocationtxs' to stakeholders
10:  c13c1f0 = 10:  c938074 psbtsigner: push the sighashtype byte after the signature
11:  d9caef9 ! 11:  e5cbe85 Implement signature sharing through the coordinator
    @@ tests/README.md: To run the server-requiring tests, pass the postgres credential
     +
     +Note that we log each daemon log, and we start them with `log_level = "trace"`.
     
    - ## tests/coordinatord (deleted) ##
    -@@
    --Subproject commit 48b67f72b948e680c78276d73b2e50dffbdddb7f
    -
      ## tests/fixtures.py ##
     @@
      """
12:  44e3dfd = 12:  a4690df bitcoind: idle, don't throttle CPU
13:  5d1fd5e = 13:  ee2a5c7 db: add a "fullysigned" column to the "transactions" table
14:  d23d58d = 14:  317ac38 bitcoind: always mark a new deposit as unconfirmed
15:  75a5e53 = 15:  ab4e92e Add a new background thread polling the coordinator for signature
```

**Comment by darosior** [comment]
So the merge of this one gave me some headaches because of the submodules, see https://github.com/re-vault/maintainer-tools/commit/491991731b9306989511b72605cf692590d92427

**Pull request merged**